src/mavlink-router/mainloop.cpp

```diff
@@ -132,12 +132,12 @@ int Mainloop::write_msg(Endpoint *e, const struct buffer *buf)
 }
 
 void Mainloop::route_msg(struct buffer *buf, int target_sysid, int target_compid, int sender_sysid,
-                         int sender_compid)
+                         int sender_compid, Endpoint *src)
 {
     bool unknown = true;
 
     for (Endpoint **e = g_endpoints; *e != nullptr; e++) {
-        if ((*e)->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid)) {
+        if (*e != src && (*e)->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid)) {
```

Comment:
That's ok, I'll move this check to accept_msg as requested and spend some more time running on it. Maybe it makes sense to add a flag / config option to enable or disable loopback?